Packages/ClientRuntime/Sources/Waiters/WaiterRetryer.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A type which tries the waited operation after a delay, then returns or throws the result.
+/// Called by the `Waiter` to make the initial request, then called again for an additional request when retry
+/// is needed.
+/// A custom closure may be injected at initialization to allow `WaiterRetryer` to be used as a mock.
+class WaiterRetryer<Input, Output> {
```

Comment:
Same as a comment I had left previously, does this need to be a class?

Reply:
It does not anymore.  Changed to `struct`.